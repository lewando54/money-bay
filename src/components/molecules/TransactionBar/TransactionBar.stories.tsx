/* eslint-disable @typescript-eslint/consistent-type-assertions */
import { Meta,StoryObj } from '@storybook/react-native'

import TransactionBar from './TransactionBar'

const meta = {
    component: TransactionBar,
    argTypes: {
        title: { control: 'text' },
        subTitle: { control: 'text' },
        amount: { control: 'number' },
        percent: { control: 'number' },
        isExpense: { control: 'boolean' }
    }
} satisfies Meta<typeof TransactionBar>

export default meta
type Story = StoryObj<typeof TransactionBar>

export const Default: Story = {
    args: {
        title: 'Adalyn Roth',
        subTitle: 'Money transfer',
        amount: 140.00,
        percent: 20,
        isExpense: true
    }
}

export const WithoutPercentSign: Story = {
    args: {
        title: 'Adalyn Roth',
        subTitle: 'Money transfer',
        amount: 140.00,
        isExpense: true
    }
}

export const IsAnIncome: Story = {
    args: {
        title: 'Adalyn Roth',
        subTitle: 'Money transfer',
        amount: 140.00,
        isExpense: false
    }
}
