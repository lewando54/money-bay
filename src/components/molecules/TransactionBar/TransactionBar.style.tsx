import { StyleSheet } from 'react-native'
export default StyleSheet.create({
    container: {
        flexDirection: 'row', 
        alignItems: 'center', 
        gap: 13
    },
    balanceContainer: {
        flexDirection: 'row', 
        marginLeft: 'auto', 
        gap: 5
    }
})
