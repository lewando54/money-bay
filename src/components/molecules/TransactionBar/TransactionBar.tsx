import React from 'react'
import { View, Text, Dimensions } from 'react-native'
import CircleWithIcon from '../../atoms/CircleWithIcon/CircleWithIcon'
import UniversalContainer from '../../atoms/UniversalContainer/UniversalContainer'
import transactionBarStyle from './TransactionBar.style'
import { MaterialIcons } from '@expo/vector-icons'
import globalThemeStyle, { GOOD_GREEN } from '../../../styling/GlobalTheme.style'

export type TTransactionsArray = Array<{
    icon: React.ReactNode,
    title: string,
    subTitle: string,
    amount: number,
    date: Date,
    percent?: number,
    isExpense?: boolean,
    testID?: string
}>

interface ITransactionBarProps {
    icon?: React.ReactNode
    title?: string
    subTitle?: string
    amount?: number
    percent?: number
    isExpense?: boolean
    testID?: string
}

export default function TransactionBar({
    icon=<MaterialIcons name="compare-arrows" size={24} color="black" />,
    title='Adalyn Roth',
    subTitle='Money transfer',
    amount=140.00,
    percent,
    isExpense=true,
    testID
}: ITransactionBarProps){
    const {width} = Dimensions.get('screen')

    return (
        <UniversalContainer padding={width * 0.0266} testID={testID} style={transactionBarStyle.container} castShadow={true}>
            <CircleWithIcon icon={icon} size={width * 0.1066}/>
            <View>
                <Text style={globalThemeStyle.text_Title}>{title}</Text>
                <Text style={globalThemeStyle.text_SubTitle}>{subTitle}</Text>
            </View>
            <View style={transactionBarStyle.balanceContainer}>
                <Text style={[globalThemeStyle.text_Title, !isExpense && {color: GOOD_GREEN}]}>{isExpense ? '-' : '+'}</Text>
                <View>
                    <Text style={[globalThemeStyle.text_Title, !isExpense && {color: GOOD_GREEN}]}>{amount.toFixed(2)}</Text>
                    <Text style={[globalThemeStyle.text_SubTitle, {marginLeft: 'auto'}, !percent && {display: 'none'}]}>{percent+'%'}</Text>
                </View>
            </View>
        </UniversalContainer>
    )
}
