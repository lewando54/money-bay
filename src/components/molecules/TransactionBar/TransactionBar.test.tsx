import React from 'react'
import renderer from 'react-test-renderer'
import TransactionBar from './TransactionBar'
describe('TransactionBar', () => {
    it('should render correctly', () => {
        const tree = renderer.create(<TransactionBar />)
        expect(tree).toMatchSnapshot()
    })

    it('should hide percent by default', () => {
        const tree = renderer.create(<TransactionBar />)
        expect(tree).toMatchSnapshot()
    })

    it('should be expense by default', () => {
        const tree = renderer.create(<TransactionBar />)
        expect(tree).toMatchSnapshot()
    })

    it('should show percent when passed', () => {
        const tree = renderer.create(<TransactionBar percent={30}/>)
        expect(tree).toMatchSnapshot()
    })

    it('should have green color of balance and + sign when isExpense is false', () => {
        const tree = renderer.create(<TransactionBar isExpense={false}/>)
        expect(tree).toMatchSnapshot()
    })

    it('should render correct text in title and subtitle', () => {
        const tree = renderer.create(<TransactionBar />)
        expect(tree).toMatchSnapshot()
    })
})
