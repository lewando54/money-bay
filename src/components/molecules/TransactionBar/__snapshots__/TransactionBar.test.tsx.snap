// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionBar should be expense by default 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "white",
        "borderRadius": 10,
        "height": "auto",
        "padding": 19.95,
        "width": "100%",
      },
      {
        "elevation": 20,
        "shadowColor": "#16196670",
        "shadowOffset": {
          "height": 6,
          "width": 0,
        },
        "shadowOpacity": 0.08,
        "shadowRadius": 16,
      },
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 13,
      },
    ]
  }
  testID="universal-container"
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#F1F5FD",
          "borderRadius": 159.9,
          "height": 79.95,
          "justifyContent": "center",
          "width": 79.95,
        },
        false,
      ]
    }
  >
    <Text />
  </View>
  <View>
    <Text
      style={
        {
          "color": "#1B1D4D",
          "fontFamily": "MulishSemiBold",
          "fontSize": 14,
        }
      }
    >
      Adalyn Roth
    </Text>
    <Text
      style={
        {
          "color": "#4C4C60",
          "fontFamily": "MulishRegular",
          "fontSize": 12,
        }
      }
    >
      Money transfer
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "gap": 5,
        "marginLeft": "auto",
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishSemiBold",
            "fontSize": 14,
          },
          false,
        ]
      }
    >
      -
    </Text>
    <View>
      <Text
        style={
          [
            {
              "color": "#1B1D4D",
              "fontFamily": "MulishSemiBold",
              "fontSize": 14,
            },
            false,
          ]
        }
      >
        140.00
      </Text>
      <Text
        style={
          [
            {
              "color": "#4C4C60",
              "fontFamily": "MulishRegular",
              "fontSize": 12,
            },
            {
              "marginLeft": "auto",
            },
            {
              "display": "none",
            },
          ]
        }
      >
        undefined%
      </Text>
    </View>
  </View>
</View>
`;

exports[`TransactionBar should have green color of balance and + sign when isExpense is false 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "white",
        "borderRadius": 10,
        "height": "auto",
        "padding": 19.95,
        "width": "100%",
      },
      {
        "elevation": 20,
        "shadowColor": "#16196670",
        "shadowOffset": {
          "height": 6,
          "width": 0,
        },
        "shadowOpacity": 0.08,
        "shadowRadius": 16,
      },
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 13,
      },
    ]
  }
  testID="universal-container"
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#F1F5FD",
          "borderRadius": 159.9,
          "height": 79.95,
          "justifyContent": "center",
          "width": 79.95,
        },
        false,
      ]
    }
  >
    <Text />
  </View>
  <View>
    <Text
      style={
        {
          "color": "#1B1D4D",
          "fontFamily": "MulishSemiBold",
          "fontSize": 14,
        }
      }
    >
      Adalyn Roth
    </Text>
    <Text
      style={
        {
          "color": "#4C4C60",
          "fontFamily": "MulishRegular",
          "fontSize": 12,
        }
      }
    >
      Money transfer
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "gap": 5,
        "marginLeft": "auto",
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishSemiBold",
            "fontSize": 14,
          },
          {
            "color": "#3EB290",
          },
        ]
      }
    >
      +
    </Text>
    <View>
      <Text
        style={
          [
            {
              "color": "#1B1D4D",
              "fontFamily": "MulishSemiBold",
              "fontSize": 14,
            },
            {
              "color": "#3EB290",
            },
          ]
        }
      >
        140.00
      </Text>
      <Text
        style={
          [
            {
              "color": "#4C4C60",
              "fontFamily": "MulishRegular",
              "fontSize": 12,
            },
            {
              "marginLeft": "auto",
            },
            {
              "display": "none",
            },
          ]
        }
      >
        undefined%
      </Text>
    </View>
  </View>
</View>
`;

exports[`TransactionBar should hide percent by default 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "white",
        "borderRadius": 10,
        "height": "auto",
        "padding": 19.95,
        "width": "100%",
      },
      {
        "elevation": 20,
        "shadowColor": "#16196670",
        "shadowOffset": {
          "height": 6,
          "width": 0,
        },
        "shadowOpacity": 0.08,
        "shadowRadius": 16,
      },
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 13,
      },
    ]
  }
  testID="universal-container"
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#F1F5FD",
          "borderRadius": 159.9,
          "height": 79.95,
          "justifyContent": "center",
          "width": 79.95,
        },
        false,
      ]
    }
  >
    <Text />
  </View>
  <View>
    <Text
      style={
        {
          "color": "#1B1D4D",
          "fontFamily": "MulishSemiBold",
          "fontSize": 14,
        }
      }
    >
      Adalyn Roth
    </Text>
    <Text
      style={
        {
          "color": "#4C4C60",
          "fontFamily": "MulishRegular",
          "fontSize": 12,
        }
      }
    >
      Money transfer
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "gap": 5,
        "marginLeft": "auto",
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishSemiBold",
            "fontSize": 14,
          },
          false,
        ]
      }
    >
      -
    </Text>
    <View>
      <Text
        style={
          [
            {
              "color": "#1B1D4D",
              "fontFamily": "MulishSemiBold",
              "fontSize": 14,
            },
            false,
          ]
        }
      >
        140.00
      </Text>
      <Text
        style={
          [
            {
              "color": "#4C4C60",
              "fontFamily": "MulishRegular",
              "fontSize": 12,
            },
            {
              "marginLeft": "auto",
            },
            {
              "display": "none",
            },
          ]
        }
      >
        undefined%
      </Text>
    </View>
  </View>
</View>
`;

exports[`TransactionBar should render correct text in title and subtitle 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "white",
        "borderRadius": 10,
        "height": "auto",
        "padding": 19.95,
        "width": "100%",
      },
      {
        "elevation": 20,
        "shadowColor": "#16196670",
        "shadowOffset": {
          "height": 6,
          "width": 0,
        },
        "shadowOpacity": 0.08,
        "shadowRadius": 16,
      },
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 13,
      },
    ]
  }
  testID="universal-container"
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#F1F5FD",
          "borderRadius": 159.9,
          "height": 79.95,
          "justifyContent": "center",
          "width": 79.95,
        },
        false,
      ]
    }
  >
    <Text />
  </View>
  <View>
    <Text
      style={
        {
          "color": "#1B1D4D",
          "fontFamily": "MulishSemiBold",
          "fontSize": 14,
        }
      }
    >
      Adalyn Roth
    </Text>
    <Text
      style={
        {
          "color": "#4C4C60",
          "fontFamily": "MulishRegular",
          "fontSize": 12,
        }
      }
    >
      Money transfer
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "gap": 5,
        "marginLeft": "auto",
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishSemiBold",
            "fontSize": 14,
          },
          false,
        ]
      }
    >
      -
    </Text>
    <View>
      <Text
        style={
          [
            {
              "color": "#1B1D4D",
              "fontFamily": "MulishSemiBold",
              "fontSize": 14,
            },
            false,
          ]
        }
      >
        140.00
      </Text>
      <Text
        style={
          [
            {
              "color": "#4C4C60",
              "fontFamily": "MulishRegular",
              "fontSize": 12,
            },
            {
              "marginLeft": "auto",
            },
            {
              "display": "none",
            },
          ]
        }
      >
        undefined%
      </Text>
    </View>
  </View>
</View>
`;

exports[`TransactionBar should render correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "white",
        "borderRadius": 10,
        "height": "auto",
        "padding": 19.95,
        "width": "100%",
      },
      {
        "elevation": 20,
        "shadowColor": "#16196670",
        "shadowOffset": {
          "height": 6,
          "width": 0,
        },
        "shadowOpacity": 0.08,
        "shadowRadius": 16,
      },
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 13,
      },
    ]
  }
  testID="universal-container"
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#F1F5FD",
          "borderRadius": 159.9,
          "height": 79.95,
          "justifyContent": "center",
          "width": 79.95,
        },
        false,
      ]
    }
  >
    <Text />
  </View>
  <View>
    <Text
      style={
        {
          "color": "#1B1D4D",
          "fontFamily": "MulishSemiBold",
          "fontSize": 14,
        }
      }
    >
      Adalyn Roth
    </Text>
    <Text
      style={
        {
          "color": "#4C4C60",
          "fontFamily": "MulishRegular",
          "fontSize": 12,
        }
      }
    >
      Money transfer
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "gap": 5,
        "marginLeft": "auto",
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishSemiBold",
            "fontSize": 14,
          },
          false,
        ]
      }
    >
      -
    </Text>
    <View>
      <Text
        style={
          [
            {
              "color": "#1B1D4D",
              "fontFamily": "MulishSemiBold",
              "fontSize": 14,
            },
            false,
          ]
        }
      >
        140.00
      </Text>
      <Text
        style={
          [
            {
              "color": "#4C4C60",
              "fontFamily": "MulishRegular",
              "fontSize": 12,
            },
            {
              "marginLeft": "auto",
            },
            {
              "display": "none",
            },
          ]
        }
      >
        undefined%
      </Text>
    </View>
  </View>
</View>
`;

exports[`TransactionBar should show percent when passed 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "white",
        "borderRadius": 10,
        "height": "auto",
        "padding": 19.95,
        "width": "100%",
      },
      {
        "elevation": 20,
        "shadowColor": "#16196670",
        "shadowOffset": {
          "height": 6,
          "width": 0,
        },
        "shadowOpacity": 0.08,
        "shadowRadius": 16,
      },
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 13,
      },
    ]
  }
  testID="universal-container"
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#F1F5FD",
          "borderRadius": 159.9,
          "height": 79.95,
          "justifyContent": "center",
          "width": 79.95,
        },
        false,
      ]
    }
  >
    <Text />
  </View>
  <View>
    <Text
      style={
        {
          "color": "#1B1D4D",
          "fontFamily": "MulishSemiBold",
          "fontSize": 14,
        }
      }
    >
      Adalyn Roth
    </Text>
    <Text
      style={
        {
          "color": "#4C4C60",
          "fontFamily": "MulishRegular",
          "fontSize": 12,
        }
      }
    >
      Money transfer
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "gap": 5,
        "marginLeft": "auto",
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishSemiBold",
            "fontSize": 14,
          },
          false,
        ]
      }
    >
      -
    </Text>
    <View>
      <Text
        style={
          [
            {
              "color": "#1B1D4D",
              "fontFamily": "MulishSemiBold",
              "fontSize": 14,
            },
            false,
          ]
        }
      >
        140.00
      </Text>
      <Text
        style={
          [
            {
              "color": "#4C4C60",
              "fontFamily": "MulishRegular",
              "fontSize": 12,
            },
            {
              "marginLeft": "auto",
            },
            false,
          ]
        }
      >
        30%
      </Text>
    </View>
  </View>
</View>
`;
